File: Algorithm/C++/1000+/1338. Reduce Array Size to The Half.cc
#include <vector>
#include <unordered_map>
#include <utility>
#include <algorithm>

class Solution {
public:
    int minSetSize(std::vector<int>& arr) {
        std::unordered_map<int, int> counter;
        for (int a: arr) {
            counter[a]++;
        }
        std::vector<std::pair<int, int>> s_counter(counter.begin(), counter.end());
        sort(s_counter.begin(), s_counter.end(),
             [](const std::pair<int, int>& a, const std::pair<int, int>& b) {
                    return a.second > b.second;
             });

        int result = 0, tmp=0, n = arr.size();
        for (auto c: s_counter) {
            result += 1;
            tmp += c.second;
            if (tmp >= (n>>1)) break;
        }
        return result;
    }
};

File: Algorithm/Python/1000+/1338. Reduce Array Size to The Half.py
from collections import Counter
class Solution(object):
    def minSetSize(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        counter = Counter(arr)
        sorted_counter = sorted(counter.items(), key = lambda x: x[1], reverse=True)
        result, curr_count = 0, 0
        for k, v in sorted_counter:
            result += 1
            curr_count += v
            if curr_count >= (len(arr)>>1):
                break
        return result
